### PR TITLE
Classifier loading state

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

Package: app-project

Toward #2381

This adds an overall app loading state that represents the loading state of the project, user, and project preferences all in one. The classifier does not load until all three are successful. There's a component that nicely represents the loading state that I generalized from the original use just on the project home page.

Light theme:
Screen Shot 2021-09-08 at 10 48 23 AM

Dark theme:
Screen Shot 2021-09-08 at 10 48 11 AM

Next iteration, we'll then use the user project preferences to determine if the user should be able to load the workflow.

# Review Checklist

## General

- [x] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by srallen** [comment]
I just noticed that `ClassifierWrapper` is missing tests for the error state render, so I'll add those, but this doesn't need to hold up an initial review.

File: packages/app-project/stores/Store.js [hunk 1]
```diff
@@ -19,6 +20,23 @@ const Store = types
   .views(self => ({
     get client () {
       return getEnv(self).client
+    },
+
+    get appLoadingState () {
```
Comment (mcbouslog) [inline packages/app-project/stores/Store.js:25]: this is great! nice straightforward way to get overall state we're looking for

**Review by mcbouslog — APPROVED** [review]
This LGTM! I forced different states by hardcoding conditionals in the `Store.js`, things looked good/as expected signed in and out

File: packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js [hunk 2]
```diff
@@ -51,22 +51,25 @@ export default function ClassifierWrapper({
...
   }
 
   try {
-    if (project.loadingState === asyncStates.success) {
+    if (appLoadingState === asyncStates.success) {
       const key = user.id || 'no-user'
       return (
         <Classifier
           authClient={authClient}
+          data-testid='classifier'
```
[the remark this answers is not in the document]
Comment (srallen) [reply packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js:72]: Oh, yes, I had thought about redoing the tests with RTL, but decided I'd do it separately later. I can remove.

[the remark this answers is not in the document]
Comment (srallen) [reply packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js:72]:
```suggestion
```

**Comment by srallen** [comment]
I forgot to add specs on the error render, so I'll try to add that in another PR.

**Pull request merged**